React/Base/RCTEventDispatcher.m

```diff
@@ -149,34 +135,39 @@ - (void)sendEvent:(id<RCTEvent>)event
   }
 
   [_observersLock unlock];
-
-  [_eventQueueLock lock];
-
-  NSNumber *eventID = RCTGetEventID(event);
-
-  id<RCTEvent> previousEvent = _events[eventID];
-  if (previousEvent) {
-    RCTAssert([event canCoalesce], @"Got event %@ which cannot be coalesced, but has the same eventID %@ as the previous event %@", event, eventID, previousEvent);
-    event = [previousEvent coalesceWithEvent:event];
+  
+  if (event.canCoalesce) {
+    [_eventQueueLock lock];
+    
+    NSNumber *eventID = RCTGetEventID(event);
+    
+    id<RCTEvent> previousEvent = _events[eventID];
+    if (previousEvent) {
+      event = [previousEvent coalesceWithEvent:event];
```

Comment:
`canCoalesce` is required, but `coalesceWithEvent` is optional, we'd better do some check? or can assume if `canCoalesce` returns `YES`, it must also implements `coalesceWithEvent`.

Reply:
Yes, not ideal but it is mentioned in the comment here https://github.com/facebook/react-native/pull/15894/files#diff-eee95b3297ff73b389d2eab419631d6dR57